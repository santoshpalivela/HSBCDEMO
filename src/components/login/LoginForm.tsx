import * as React from "react";
import { LoginEntity } from "../../types/index";

import { Button, Form, FormGroup, Label, Input } from 'reactstrap';



interface PropsForm {
  onLogin: () => void;
  onUpdateField: (name: string, value: any) => void;
  loginInfo: LoginEntity;
}


  const LoginForm = (props: PropsForm) => {

  const { onLogin, onUpdateField, loginInfo } = props;


  const onTexFieldChange = fieldId => e => {
    onUpdateField(fieldId, e.target.value);
  };

  return (
    <div className="">
      
    <Form className="login-form">
      <FormGroup>
        <Label for="exampleEmail">Username</Label>
        <Input type="email" name="username" id="username" placeholder="Username" 
        value={loginInfo.login}
        onChange={onTexFieldChange("login")}/>
      </FormGroup>
      <FormGroup>
        <Label for="Password">Password</Label>
        <Input type="password" name="password" id="password" placeholder="Password" 
         value={loginInfo.password}
        onChange={onTexFieldChange("password")}
        />
      </FormGroup>
      
      <Button variant="contained" id="btnSubmit" className="btnSubmit" color="primary" onClick={onLogin}>Submit</Button>
    </Form>
    </div>
  );
};


export default LoginForm;