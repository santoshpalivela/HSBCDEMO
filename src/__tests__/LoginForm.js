import React from "react";
import { shallow, mount, render, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import LoginForm from "../components/login/LoginForm";
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() });

describe("should render LoginForm without throwing an error", () => {

  const loginInfo = { login: "", password: "" };

  const tree = renderer.create(<LoginForm loginInfo={loginInfo} />).toJSON();
  expect(tree).toMatchSnapshot();

  const mockFn = jest.fn();
  const LoginFormMounted = shallow(
    <LoginForm loginInfo={loginInfo} onLogin={mockFn} onUpdateField={mockFn} />
  );

  it("Find the  render of Login form ", () => {
    expect(LoginFormMounted.find(".login-form").exists()).toBe(true)
  });

  it("Find the  render of username test case ", () => {
    expect(LoginFormMounted.find("#username").exists()).toBe(true);
    
  });

  it("testing text change on the usename filed", () => { 
      const username="admin";   
      
    LoginFormMounted.find("#username").simulate("change", {
      target: { value: username },
    });
    expect(mockFn.mock.calls[0][0]).toBe("login");
    expect(mockFn.mock.calls[0][1]).toBe(username);
  });

  it("Find the  render of password field ", () => {
    expect(LoginFormMounted.find("#password").exists()).toBe(true);
    
  });

  it("testing text change on the password filed", () => {    
    const password="Admin123";
    LoginFormMounted.find("#password").simulate("change", {
      target: { value: password },
    });
    expect(mockFn.mock.calls[1][0]).toBe("password");
    expect(mockFn.mock.calls[1][1]).toBe(password);
  });


  it("Find the  render of submit Button ", () => {
    expect(LoginFormMounted.find("#btnSubmit").exists()).toBe(true);
    
  });

  it("clicking the submit button", () => {    
    LoginFormMounted.find("#btnSubmit").simulate("click");

    expect(JSON.stringify(mockFn.mock.calls.length)).toEqual("3");
    
  });

  
});
