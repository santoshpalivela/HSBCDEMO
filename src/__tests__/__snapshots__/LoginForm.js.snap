// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className=""
>
  <form
    className="login-form"
  >
    <div
      className="form-group"
    >
      <label
        className=""
        htmlFor="exampleEmail"
      >
        Username
      </label>
      <input
        className="form-control"
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className=""
        htmlFor="Password"
      >
        Password
      </label>
      <input
        className="form-control"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <button
      aria-label={null}
      className="btnSubmit btn btn-primary"
      id="btnSubmit"
      onClick={[Function]}
      variant="contained"
    >
      Submit
    </button>
  </form>
</div>
`;
